components/form/signUpForm.tsx

```diff
@@ -0,0 +1,181 @@
+import React from 'react'
+import { useFormik, Field } from 'formik'
+import { Button } from '../dataComponents/button'
+import { PageType } from '../Types'
+
+function validateForm(values: {
+  firstName: any
+  lastName: any
+  email: any
+  phone?: string
+  text?: string
+}) {
+  let errors: { [k: string]: any } = {}
+  if (!values.email) {
+    errors.email = 'Required'
+  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
+    errors.email = 'Invalid email format'
+  }
+  if (!values.firstName) {
+    errors.firstName = 'Required'
+  }
+  if (!values.lastName) {
+    errors.lastName = 'Required'
+  }
+  if (!values.phone) {
+    errors.phone = 'Required'
+  } else if (
+    !values.phone.match(
+      /^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$/im,
+    )
+  ) {
+    errors.phone = 'Invalid number'
+  }
+
+  if (errors.email || errors.firstName || errors.lastName || errors.phone) {
+    return errors
+  }
+  return null
+}
+
+export default function SignupForm({ data }: PageType) {
+  const dir = data.homepage.aboutTitle.includes('b') ? 'right-40' : 'left-40'
+  console.log(dir)
+  const formik = useFormik({
+    initialValues: {
+      firstName: '',
+      lastName: '',
+      email: '',
+      phone: '',
+      text: '',
+    },
+    onSubmit: (values) => {
+      let res = validateForm(values)
+
+      if (res) {
+        let firstNameError = res.firstName ? res.firstName : null
+        let lastNameError = res.lastName ? res.lastName : null
+        let emailError = res.email ? res.email : null
+        let phoneError = res.phone ? res.phone : null
+
+        alert(
+          'Error!\n \n' +
+            (firstNameError ? `first name: ${firstNameError}\n` : '') +
+            (lastNameError ? `last name: ${lastNameError}\n` : '') +
+            (emailError ? `email: ${emailError}\n` : '') +
+            (phoneError ? `phone: ${phoneError}` : ''),
+        )
+      } else {
+        alert('Form submitted successfully !')
+      }
+    },
+  })
+
+  return (
+    <form
+      onSubmit={formik.handleSubmit}
+      className={`mt-12 md:mt-0 grid p-2 justify-items-center text-3xl font-bold bg-[#FFFFFF] border border-[#FFFFFF] mt-16
+      md:absolute md:-top-20 md:w-2/5 
+       lg:absolute lg:-top-24 lg:w-2/5 
+        shadow-[0px_0px_4px_rgba(0,0,0,0.25)] box-border boreder-solid rounded-3xl md:${dir} lg:${dir}`}
```

Comment:
what is this: md:${dir} lg:${dir}?

Reply:
tried to control the side of the form 
english - to the right
hebrew - to the left
because its have 'absolute' , i see its not the best solution so i'll find another way :)

Reply:
you can do somehitng like: className={`${dir === "ltr" ? "left-0" : "right-0"}`}

Reply:
fixed, 
'absolute' removed